includes/admin/class-wc-admin-dashboard.php

```diff
@@ -24,7 +24,7 @@ class WC_Admin_Dashboard {
 		 */
 		public function __construct() {
 			// Only hook in admin parts if the user has admin access.
-			if ( current_user_can( 'view_woocommerce_reports' ) || current_user_can( 'manage_woocommerce' ) || current_user_can( 'publish_shop_orders' ) ) {
+			if ( $this->should_display_widget() && ( current_user_can( 'view_woocommerce_reports' ) || current_user_can( 'manage_woocommerce' ) || current_user_can( 'publish_shop_orders' ) ) ) {
```

Comment:
all of this logic could be moved into `$this->should_display_widget()` and simplified. Not required though.

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/29174/commits/d013d7f7659245960e051bcb4138bf36b210e2ec